app/subapps/browser/views/charm.js

```diff
@@ -354,18 +353,15 @@ YUI.add('subapp-browser-charmview', function(Y) {
         this._renderCharmView(this.get('entity'));
         this.hideIndicator(this.get('renderTo'));
       } else {
-        this.get('store').charm(this.get('entityId'), {
-          'success': function(data) {
-            var charm = new models.Charm(data.charm);
-            if (data.metadata) {
-              charm.set('metadata', data.metadata);
-            }
-            this.set('charm', charm);
-            this._renderCharmView(charm);
-            this.hideIndicator(this.get('renderTo'));
-          },
-          'failure': this.apiFailure
-        }, this);
+        this.get('charmstore').getEntity(
+            this.get('entityId'),
+            function(charms) {
+              var charm = charms[0];
```

Comment:
I'm a little confused on what you're getting back on getEntity and how we know we always just want the first one.